proxyd/rewriter.go

```diff
@@ -89,9 +91,9 @@ func rewriteParam(rctx RewriteContext, req *RPCReq, res *RPCRes, pos int) (Rewri
 
 	// we assume latest if the param is missing,
 	// and we don't rewrite if there is not enough params
-	if len(p) == pos {
+	if len(p) == pos && !required {
```

Comment:
What does this mean?

Reply:
it differentiates if the param is required or not

when the param is optional, we assume `latest` as the default value
when it is required, we just prevent rewriting anything